### PR TITLE
feat(proxy): introduce Concurrency under EnvoyProxy resource

### PR DESCRIPTION
**Pull request by ardikabs**
**What type of PR is this?**
Introduce an additional field under `EnvoyProxy` for additional proxy-level configuration options. The idea should be similar to Istio's [ProxyConfig](https://istio.io/latest/docs/reference/config/networking/proxy-config/).

**What this PR does / why we need it**: Our use case demands configuring the non-default concurrency for the Envoy proxy which currently is not supported by Envoy Gateway.

**Which issue(s) this PR fixes**: #1718 

Fixes #

### CODE REVIEW

**Comment by ardikabs** [comment]
cc: @zirain

File: api/config/v1alpha1/envoyproxy_types.go [hunk 1]
```diff
@@ -57,6 +57,19 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Bootstrap *string `json:"bootstrap,omitempty"`
+
+	// Config defines proxy level configuration options
+	//
+	// +optional
+	Config *ProxyConfig `json:"config,omitempty"`
```
Comment (arkodg) [inline api/config/v1alpha1/envoyproxy_types.go:64]: imo lets rm the top level `Config` field and directly add `Concurrency` here

Comment (ardikabs) [reply api/config/v1alpha1/envoyproxy_types.go:64]: @arkodg actually, I'm fine either moving Concurrency to the top level or keeping it within the Config field as both satisfy my use case, but should we also take future extensibility into account to accommodate other proxy-level options when needed, wdyt?

Comment (arkodg) [reply api/config/v1alpha1/envoyproxy_types.go:64]: this field lives inside `EnvoyProxy.Spec` so we already have a envoy proxy specific home created, other fields at the same level are `logging` & `bootstrap` which are also envoy proxy cmdline args

Comment (ardikabs) [reply api/config/v1alpha1/envoyproxy_types.go]: make sense, fixed.

File: api/config/v1alpha1/envoyproxy_types.go [hunk 2]
```diff
@@ -57,6 +57,12 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Bootstrap *string `json:"bootstrap,omitempty"`
+
+	// Concurrency defines the number of worker threads to run. If unset, the default back to
```
Comment (arkodg) [inline api/config/v1alpha1/envoyproxy_types.go:61]:
```suggestion
	// Concurrency defines the number of worker threads to run. If unset, it defaults to
```

File: api/config/v1alpha1/envoyproxy_types.go [hunk 3]
```diff
@@ -57,6 +57,12 @@ type EnvoyProxySpec struct {
...
+	// Envoy default value which the number of hardware threads on the machine.
```
Comment (arkodg) [inline api/config/v1alpha1/envoyproxy_types.go:62]:
```suggestion
	// the the number of cpuset threads on the platform.
```

File: internal/infrastructure/kubernetes/proxy/resource.go [hunk 4]
```diff
@@ -143,10 +143,16 @@ func expectedProxyContainers(infra *ir.ProxyInfra, deploymentConfig *egcfgv1a1.K
 		fmt.Sprintf("--log-level %s", logLevel),
 		"--cpuset-threads",
 	}
+
 	if componentLogLevel := componentLogLevelArgs(proxyLogging.Level); componentLogLevel != "" {
 		args = append(args, fmt.Sprintf("--component-log-level %s", componentLogLevel))
 	}
 
+	if infra.Config != nil &&
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/proxy/resource.go:151]: can we keep this closer to L144 near `cpuset-threads` ?

Comment (ardikabs) [reply internal/infrastructure/kubernetes/proxy/resource.go:147]: okay, done!

**Comment by ardikabs** [comment]
@arkodg yes, i missed the `make generate`. Fixed!

**Review by arkodg — APPROVED** [review]
LGTM thanks !
lets wait until v0.5 lands to merge this PR, since this PR introduces doc changes

**Review by zirain — APPROVED** [review]

**Pull request merged**